File: app/src/main/java/com/personal/vamsi/wayto/RouteFragment.java
package com.personal.vamsi.wayto;

import android.app.TimePickerDialog;
import android.content.Intent;
import android.content.res.AssetManager;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.text.InputType;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.AutoCompleteTextView;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TimePicker;

import java.util.Calendar;
import java.util.TimeZone;


public class RouteFragment extends Fragment {
    JsonExtracter2.Buildings buildings;
    public RouteFragment(){

    }
    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        AssetManager assetManager = getResources().getAssets();
         buildings = new JsonExtracter2().doInBackground(assetManager);
    }



    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        String[] modes = {"driving","walking","transit","bicycling"};
        View view = inflater.inflate(R.layout.route_fragment, container, false);
        final AutoCompleteTextView fromView = (AutoCompleteTextView) view.findViewById(R.id.From);
        ArrayAdapter<String> adapter = new ArrayAdapter<String>(getActivity(),android.R.layout.simple_list_item_1,
                buildings.searchText);
         ArrayAdapter<String> modearray = new ArrayAdapter<String>(getActivity(),android.R.layout.simple_list_item_1, modes);

        fromView.setAdapter(adapter);

        final AutoCompleteTextView toView = (AutoCompleteTextView) view.findViewById(R.id.To);
        toView.setAdapter(adapter);

        final AutoCompleteTextView mode = (AutoCompleteTextView) view.findViewById(R.id.mode);
        mode.setAdapter(modearray);
        Calendar calendar = Calendar.getInstance(TimeZone.getTimeZone("America/New_York"));
        EditText searchTime = (EditText)view.findViewById(R.id.time);
        searchTime.setInputType(InputType.TYPE_NULL);

        TimePickerDialog timePickerDialog = new TimePickerDialog(getActivity(), new TimePickerDialog.OnTimeSetListener() {

            @Override
            public void onTimeSet(TimePicker view, int hourOfDay, int minute) {
                Calendar calendar = Calendar.getInstance(TimeZone.getTimeZone("America/New_York"));
                calendar.set(Calendar.HOUR_OF_DAY, hourOfDay);
                calendar.set(Calendar.MINUTE, minute);

            }
        },calendar.get(Calendar.HOUR_OF_DAY),calendar.get(Calendar.MINUTE), true);
        Button directions = view.findViewById(R.id.button2);
        directions.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String[] fromBuil = buildings.searchText;
                Intent intent = new Intent(getActivity(), MapsActivityRoute.class);

                for(int i=0;i<fromBuil.length;i++)
                {
                    if(fromBuil[i].equals(fromView.getText().toString())){
                        intent.putExtra("name",buildings.name[i]);
                        intent.putExtra("fromlat",buildings.locations[i].latitude);
                        intent.putExtra("fromlong",buildings.locations[i].longitude);
                        intent.putExtra("mode",mode.getText().toString());
                    }
                }
                 for(int i=0;i<fromBuil.length;i++)
                {
                    if(fromBuil[i].equals(toView.getText().toString())){
                        intent.putExtra("toname",buildings.name[i]);
                        intent.putExtra("tolat",buildings.locations[i].latitude);
                        intent.putExtra("tolong",buildings.locations[i].longitude);
                    }
                }

                startActivity(intent);
            }
        });
        return view;
    }




}

File: app/src/main/java/com/personal/vamsi/wayto/RouteFragment.kt
package com.personal.vamsi.wayto

import android.app.TimePickerDialog
import android.content.Intent
import android.content.res.AssetManager
import android.os.Bundle
import android.support.v4.app.Fragment
import android.text.InputType
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.*

import java.util.Calendar
import java.util.TimeZone


class RouteFragment : Fragment() {
    internal var buildings: JsonExtracter.Buildings?=null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val assetManager = resources.assets
        buildings = JsonExtracter().doInBackground(assetManager).buildings
    }


    @Suppress("DEPRECATION")
    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        // Inflate the layout for this fragment
        val modes = arrayOf("driving", "walking", "transit", "bicycling")
        val view = inflater!!.inflate(R.layout.route_fragment, container, false)
        val fromView = view.findViewById<View>(R.id.From) as AutoCompleteTextView
        var locationSuggestions = buildings?.searchText!!.plus("My location")

        // array adapter for from textview
        val fromadapter: ArrayAdapter<String>? =
                ArrayAdapter(
                        activity,
                android.R.layout.simple_list_item_1,
                locationSuggestions)
        // array adapter for to textview
        val toadapter: ArrayAdapter<String>? =
                ArrayAdapter(
                        activity,
                android.R.layout.simple_list_item_1,
                        buildings?.searchText!!)
        // array adapter for mode textView
        val modearray = ArrayAdapter(activity, android.R.layout.simple_list_item_1, modes)

        fromView.setText("From: My location")
        fromView.setAdapter(fromadapter)

        val toView = view.findViewById<View>(R.id.To) as AutoCompleteTextView
        toView.setAdapter(toadapter)

        val mode = view.findViewById<View>(R.id.mode) as AutoCompleteTextView
        mode.setAdapter(modearray)



        var calendar = Calendar.getInstance(TimeZone.getTimeZone("America/New_York"))
        val timeclick = view.findViewById<EditText>(R.id.timeView)
        timeclick.setText("${calendar.get(Calendar.HOUR_OF_DAY)}:${calendar.get(Calendar.MINUTE)}")

        val timePickerDialog = TimePickerDialog(activity, TimePickerDialog.OnTimeSetListener { view, hourOfDay, minute ->
            //val calendar = Calendar.getInstance(TimeZone.getTimeZone("America/New_York"))
            calendar.set(Calendar.HOUR_OF_DAY, hourOfDay)
            calendar.set(Calendar.MINUTE, minute)
            //Log.v("time in am and pm",""+calendar.timeInMillis)
            timeclick.setText("${calendar.get(Calendar.HOUR_OF_DAY)}:${calendar.get(Calendar.MINUTE)}")
        }, calendar.get(Calendar.HOUR_OF_DAY), calendar.get(Calendar.MINUTE), false)

        timeclick.setOnClickListener {
            timePickerDialog.show()
        }

        val directions = view.findViewById<Button>(R.id.button2)
        directions.setOnClickListener {
            val fromBuil = buildings!!.searchText
            val intent = Intent(activity, MapsActivity::class.java)
            intent.putExtra("action","navigation")
            if (fromView.text.toString().equals("From: My location")||fromView.text.toString().equals("My location")){
                intent.putExtra("fromname","My location")
            }
            else if(fromBuil!!.contains(fromView.text.toString())){
                intent.putExtra("fromname", buildings!!.name!![fromBuil!!.indexOf(fromView.text.toString())])
                intent.putExtra("fromlat", buildings!!.locations!![fromBuil!!.indexOf(fromView.text.toString())].latitude)
                intent.putExtra("fromlong", buildings!!.locations!![fromBuil!!.indexOf(fromView.text.toString())].longitude)
            }
            else{
                Toast.makeText(context,"Invalid from location. Using CURRENT GPS LOCATION!!!",Toast.LENGTH_LONG).show()
                intent.putExtra("fromname","My location")
            }

            if(fromBuil!!.contains(toView.text.toString())) {
                intent.putExtra("toname", buildings!!.name!![fromBuil!!.indexOf(toView.text.toString())])
                intent.putExtra("tolat", buildings!!.locations!![fromBuil!!.indexOf(toView.text.toString())].latitude)
                intent.putExtra("tolong", buildings!!.locations!![fromBuil!!.indexOf(toView.text.toString())].longitude)

            }
            else{
                Toast.makeText(context,"Invalid 'To Location'. Choosing UGA LOCATION!!!",Toast.LENGTH_LONG).show()
                intent.putExtra("toname", "UGA")
                intent.putExtra("tolat", 33.9480053)
                intent.putExtra("tolong",-83.3773221)
            }
            if (modes.contains(mode.text.toString())){
                intent.putExtra("mode", mode.text.toString())
                intent.putExtra("hours",calendar.get(Calendar.HOUR_OF_DAY))
                intent.putExtra("mins",calendar.get(Calendar.MINUTE))
                //intent.putExtra("hours",calendar.get())
                startActivity(intent)
            }
            else{
                Toast.makeText(context,"Invalid 'Transport Mode'. Using driving as default!!!",Toast.LENGTH_LONG).show()
                intent.putExtra("mode","driving")
                intent.putExtra("hours",calendar.get(Calendar.HOUR_OF_DAY))
                intent.putExtra("mins",calendar.get(Calendar.MINUTE))
                startActivity(intent)
            }

        }
        return view
    }


}
